New pool lib

lvn-rusty-dragon opened the pull request.

lvn-rusty-dragon committed:
Initial move to bb8

lvn-rusty-dragon committed:
Idle timeouts

lvn-rusty-dragon committed:
Begin new approach to performing queries

lvn-rusty-dragon committed:
Convert to new query system

lvn-rusty-dragon committed:
Remove no longer needed mutexes

lvn-rusty-dragon committed:
Retry and timeout logic

lvn-rusty-dragon committed:
Just one perform_query

lvn-rusty-dragon committed:
Timeout kills a connection

lvn-rusty-dragon committed:
Relax thiserror dep

lvn-rusty-dragon committed:
Include error in warnings

lvn-hasky-dragon commented on packages/cosmos/src/client.rs:
```diff
@@ -67,18 +69,23 @@ impl CosmosBuilders {
 
-    async fn recycle(&self, _: &mut CosmosInner) -> RecycleResult<anyhow::Error> {
+    async fn is_valid(&self, _: &mut CosmosInner) -> Result<()> {
+        // We do the validity check within the perform_query calls and update is_broken
```
Given that this function is used internally by the pool library (Ref: https://github.com/djc/bb8/blob/6908d2f894dd2ed837538c0bb0fd06c2a37d5928/bb8/src/inner.rs#L125), do you think something like this is better:

```
if !inner.is_broken { Ok()} else anyhow!("Some failure")
```

lvn-rusty-dragon replied:
Yeah, that makes sense, I'll update it. I would have thought it would check is_broken right there too based on the docs.

lvn-rusty-dragon committed:
clippy

lvn-rusty-dragon committed:
Better is_valid

lvn-rusty-dragon committed:
Fix regression in public key

Not related to current PR, but blocking it from being used. Previously
we wanted to allow simulating transactions without having a signing key.
Accidentally used that on actual signing too.
